src/android/CryptoUtils.java

```
@@ -0,0 +1,127 @@
+            byte[] encrypted = cipher.doFinal(clearText.getBytes(StandardCharsets.UTF_8));
+            return String.format("%s%s%s", Crypto.toBase64(iv), DELIMITER, Crypto.toBase64(encrypted));
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static String decrypt(String ciphertext) {
+        String[] fields = ciphertext.split(DELIMITER);
+        if (fields.length != 2) {
+            throw new IllegalArgumentException("Invalid encypted text format");
+        }
+        byte[] iv = Crypto.fromBase64(fields[0]);
+        byte[] cipherBytes = Crypto.fromBase64(fields[1]);
+
+        SecretKey secretKey = null;
+        try {
+            secretKey = getKey();
+            final Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+            final GCMParameterSpec spec = new GCMParameterSpec(128, iv);
+            cipher.init(Cipher.DECRYPT_MODE, secretKey, spec);
+            final byte[] decodedData = cipher.doFinal(cipherBytes);
+            final String unencryptedString = new String(decodedData, StandardCharsets.UTF_8);
+            return unencryptedString;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static boolean isCypherText(String value) {
+        return value.contains(DELIMITER);
+    }
+
+    public static boolean isKeyHardwareBacked() {
+        try {
+            final KeyStore keyStore = KeyStore.getInstance(ANDROID_KEY_STORE_PROVIDER);
+            keyStore.load(null);
+            SecretKey key = (SecretKey) keyStore.getKey(KEY_ALIAS, null);
+            SecretKeyFactory factory = SecretKeyFactory.getInstance(key.getAlgorithm(), ANDROID_KEY_STORE_PROVIDER);
+            KeyInfo info = (KeyInfo) factory.getKeySpec(key, KeyInfo.class);
+            return info.isInsideSecureHardware();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return false;
```

maybe throw an error instead of catching